### PR TITLE
Move choropleth to class

### LINKED ISSUE #981: Toggle or Show parameter for choropleth layers not working as expected
`#`### Please add a code sample or a nbviewer link, copy-pastable if possible

```python
m.choropleth(name = 'Registered Electric Vehicles',geo_data = zips, data = zips, key_on='feature.properties.ZIP',
                     columns=['ZIP','EV'], fill_color='BuGn', line_weight = 0.3, line_opacity = 0.5,
                     fill_opacity = 0.5, legend_name = 'Registered Electric Vehicles', 
                     smooth_factor = 1, toggle = True)

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-54-828389e26172> in <module>()
     11                      columns=['ZIP','PHEV'], fill_color='RdPu', line_weight = 0.3, line_opacity = 0.5,
     12                      fill_opacity = 0.5, legend_name = 'Registered Plug-in Hybrids',
---> 13                      smooth_factor = 1, toggle = True)
     14 m.choropleth(name = 'Registered Electric Vehicles',geo_data = zips, data = zips, key_on='feature.properties.ZIP',
     15                      columns=['ZIP','EV'], fill_color='BuGn', line_weight = 0.3, line_opacity = 0.5,

TypeError: choropleth() got an unexpected keyword argument 'toggle'
```
#### Problem description
I have a map with multiple choropleth layers that I want available to select in LayerControl, but I only want one to display on load. 

Based on #772 and #606 it seems like anything inheriting the `Layer` class should have the parameter `toggle` to enable or disable the layer on load. This doesn't work for choropleths, which I had thought inherited the Layer class. I've also tried adding the choropleth to a `FeatureGroup` with `show = False`, but this doesn't work because choropleth is a method of a `Map` object.  

What am I doing wrong? or is there not currently a way to only display one choropleth layer on load?

#### Output of ``folium.__version__``
0.6.0

### PR DESCRIPTION
**Pull request by Conengmo**
Closes #981.

I have been meaning to do this for a while now :) Moving the `Map.choropleth()` method to its own class, so it's similar to all other folium types. This is not a super important change, but I think it makes our code base more consistent, which is good for developers, and makes our API more consistent, which is good for users.

I haven't changed any functionality, except:

* `Choropleth` accepts `show`, `overlay` and `control` arguments like other layer types.
* Made `Choropleth` a subclass of `FeatureGroup` and tweaked the rendering.
* Removed the `reset` option, since it didn't do anything.
* `Map.choropleth` method still works but raises a `FutureWarning`

Note about `ColorMap`:
It needs `Map` as its parent, or we'd have to change branca. Also it cannot be included in the `FeatureGroup`, so it cannot be toggled in layer control.

Links to the changed notebooks:
[Quickstart.ipynb](https://nbviewer.jupyter.org/github/conengmo/folium/blob/choropleth-refactor/examples/Quickstart.ipynb)
[GeoJSON_and_choropleth.ipynb](https://nbviewer.jupyter.org/github/conengmo/folium/blob/choropleth-refactor/examples/GeoJSON_and_choropleth.ipynb)

I made sure the tests still work and updated the example notebooks. Didn't add a line to the changelog yet.

@ocefpaf or @jtbaker, want to review this one?

### CODE REVIEW

File: folium/features.py [hunk 1]
```diff
@@ -722,6 +725,248 @@ def render(self, **kwargs):
...
+    ...            topojson='objects.countries')
+    >>> Choropleth(geo_data='geo.json', data=df,
+    ...            columns=['Data 1', 'Data 2'],
+    ...            key_on='feature.properties.myvalue',
+    ...            fill_color='PuBu',
+    ...            bins=[0, 20, 30, 40, 50, 60],
+    ...            highlight=True)
+    """
+
+    def __init__(self, geo_data, data=None, columns=None, key_on=None,
```
[the remark this answers is not in the document]
Comment (Conengmo) [reply folium/features.py]: Yeah it's complex :p

[the remark this answers is not in the document]
Comment (Conengmo) [reply folium/features.py]: Definitely complex, but it'll have to do

**Comment by Conengmo** [comment]
Thanks for the review and merge @ocefpaf. I agree it's a bit of an outlier but it seems much used given the number of issues we get on it. 

By the way, the commits weren't squashed. Did the default setting changed? I'm not paying much attention to my commits since I thought we're squashing .

**Comment by ocefpaf** [comment]
> Did the default setting changed?

Probably. I did not noticed that :-/

> 'm not paying much attention to my commits since I thought we're squashing .

Same here. Not a big deal though. I'll take a better look next time before merging.

**Pull request merged**